CRM/MembershipExtras/Hook/ValidateForm/MembershipType.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * Class CRM_MembershipExtras_Hook_ValidateForm_MembershipType
+ *
+ * Validates Membership Type Form
+ */
+class CRM_MembershipExtras_Hook_ValidateForm_MembershipType {
+
+  /**
+   * @var array
+   *   List of the submitted fields and their values passed from the hook.
+   */
+  private $fields;
+
+  /**
+   * @var array
+   *   List of form validation errors passed from the hook.
+   */
+  private $errors;
+
+  /**
+   * CRM_MembershipExtras_Hook_ValidateForm_MembershipType constructor.
+   *
+   * @param $fields
+   * @param $errors
+   */
+  public function __construct(&$fields, &$errors) {
+    $this->fields = &$fields;
+    $this->errors = &$errors;
+  }
+
+  /**
+   * Validates the membership form submission
+   */
+  public function validate() {
+    $membershipPeriodType = $this->fields['period_type'];
+    if ($membershipPeriodType == 'rolling') {
+      return;
+    }
+
+    $durationInterval = $this->fields['duration_interval'];
```

Comment:
I think it's better to move code from line 42 and below to new private method and add docblock explaining what kind of validation it does.